core/domain/exp_services.py

```diff
@@ -1586,9 +1563,6 @@ def load_demo(exploration_id):
     """
     delete_demo(exploration_id)
 
-    if not exp_domain.Exploration.is_demo_exploration_id(exploration_id):
```

Comment:
Ditto.

Reply:
Load demo calls delete_demo and there exists the same exact check. See here:
https://github.com/oppia/oppia/blob/7ad9d06e434c589f0aaa015252ca65872557b7eb/core/domain/exp_services.py#L1563

Reply:
I think we should call `is_demo_exploration_id` before calling `delete_demo`. will that be fine?

Reply:
done